### DISCUSSION TITLE
PowerShell Scripts

### DISCUSSION THREAD

**Discussion by jamesshenry** [alirezanet/Husky.Net#67] [General]
Is it possible to use PowerShell scripts instead of csx?

**Comment by alirezanet** [comment]
Hi @henry-js, 
Yes, it is possible, although running PowerShell scripts might be a little bit tricky since you need to make sure everyone using your repo has set the execution policy. 
here is one example using PowerShell 7:

task-runner.json
```json
   "tasks": [
      {
         "name": "ps",
         "command": "pwsh",
         "args": [
            ".husky/hello-world.ps1"
         ]
      },
```
to test it:
```
dotnet husky run --name ps
```

**Reply by alirezanet** [reply]
But if you mean directly using the `exec` command, the answer is no, however might be a cool feature to add 🤔

**Reply by jamesshenry** [reply]
That's great! Works well enough as-is. What benefits would running it via exec provide?

**Reply by alirezanet** [reply]
Right now, you have to make sure all of the contributors can run the PowerShell script on their computers because the execution policy on windows by default is restricted. also, not everyone has PowerShell installed on their computer especially Linux and Mac users. (This only could be a problem when you have multiple contributors)
